### PR TITLE
[IOSP-453] Add VerticalFrameworks.md

### PR DESCRIPTION
**Pull request by inamiy**
Ticket: https://babylonpartners.atlassian.net/browse/IOSP-453

This PR adds `VerticalFrameworks` guideline that explains how to organise each squad's vertical framework and what code can be shared in `BabylonVerticalInterfaces.framework`.

### Note

Section `2. What are the rules for each Vertical Framework?` is not complete yet as vertical-management-related topics e.g. Current, Feature switch, localisation, are not fully documented yet.
They will be created/modified in a separated document.

### CODE REVIEW

File: Cookbook/Technical-Documents/VerticalFrameworks.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+After modularisation refactoring in [IOSP-338](https://babylonpartners.atlassian.net/browse/IOSP-338), our Babylon App's UI screen framework dependencies are now fully decoupled from each other so called **Vertical Frameworks**.
+It ensures that there are **no dependencies between each vertical** so that faster parallel build and easy maintainance become possible.
+
+<img src="Assets/parallel-build.png" width="500"><br>
+(Example of parallel build in Vertical Frameworks)<br>
+
+
+In order to share each vertical's functionalities among other verticals, e.g. `ClinicalRecordsUI` using `MapsUI` feature, rather than one depending on the other sequentially, we now add **`BabylonVerticalInterfaces.framework`** ([PR-9964](https://github.com/babylonhealth/babylon-ios/pull/9964)) as a shim to **share each vertical's interfaces only** to the other verticals.
+
+<img src="Assets/dependency-v4.12.0.png">
```
Comment (sergdort) [inline Cookbook/Technical-Documents/VerticalFrameworks.md]: It does not show image. Probably we should use

Comment (inamiy) [reply Cookbook/Technical-Documents/VerticalFrameworks.md]: It's visible in https://github.com/babylonhealth/ios-playbook/blob/94a94840c3f5a52175ac64923dbac3008c0db6d0/Cookbook/Technical-Documents/VerticalFrameworks.md , so should be no problem.

**Review by sergdort — APPROVED** [review]

**Review by AliSoftware — COMMENTED** [review]
Nice doc overall! 👍 

Wondering if we should maybe add a paragraph to say that not all protocols _need_ to be moved to `VerticalInterfaces`, but _only_ the ones corresponding to features that are intended/needed to be shared across verticals. Though that probably goes without saying 🤷‍♂ (just don't want people starting to migrate all the existing protocols there if they are only used by their own vertical)

File: Cookbook/Technical-Documents/VerticalFrameworks.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+2. What are the rules for each Vertical Framework?
+
+## 1. What goes into `BabylonVerticalInterfaces` ?
+
+Just like C's header file, `BabylonVerticalInterfaces` must only contain interfaces that describes the **syntax of the API** without exposing its implementation (semantics).
+
+Following code are considered as shareable interfaces:
+
+1. `protocol`s (without default implementation)
+2. `struct`s and `enum`s that protocol-method uses (DTOs)
```
[the remark this answers is not in the document]
Comment (inamiy) [reply Cookbook/Technical-Documents/VerticalFrameworks.md]:
```suggestion
2. `struct`s and `enum`s that those protocols use (DTOs)
```

Will use suggestion in https://github.com/babylonhealth/ios-playbook/pull/303#discussion_r357255396 instead.

**Review by AliSoftware — APPROVED** [review]

File: Cookbook/Technical-Documents/VerticalFrameworks.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+## Overview
+
+After modularisation refactoring in [IOSP-338](https://babylonpartners.atlassian.net/browse/IOSP-338), our Babylon App's UI screen framework dependencies are now fully decoupled from each other so called **Vertical Frameworks**.
+It ensures that there are **no dependencies between each vertical** so that faster parallel build and easy maintainance become possible.
+
+<img src="Assets/parallel-build.png" width="500"><br>
+(Example of parallel build in Vertical Frameworks)<br>
+
+
+In order to share each vertical's functionalities among other verticals, e.g. `ClinicalRecordsUI` using `MapsUI` feature, rather than one depending on the other sequentially, we now add **`BabylonVerticalInterfaces.framework`** ([PR-9964](https://github.com/babylonhealth/babylon-ios/pull/9964)) as a shim to **share each vertical's interfaces only** to the other verticals.
```
[the remark this answers is not in the document]
Comment (inamiy) [reply Cookbook/Technical-Documents/VerticalFrameworks.md]:
```suggestion
This means that from now on, if a vertical needs to use a functionality from another vertical (e.g. `ClinicalRecordsUI` using `MapsUI` feature), we must use **`BabylonVerticalInterfaces.framework`** ([PR-9964](https://github.com/babylonhealth/babylon-ios/pull/9964)) as a shim to **share each vertical's interfaces only** to the other verticals. This is for keeping verticals independent from each other to make parallel build possible.
```

@AliSoftware https://github.com/babylonhealth/ios-playbook/pull/303#discussion_r357249855
Simplified a bit.

**Pull request merged**